app/bundles/LeadBundle/Event/CompanyBuilderEvent.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Event;
+
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\Process\Exception\InvalidArgumentException;
+
+/**
+ * Class CompanyBuilderEvent
+ */
+class CompanyBuilderEvent extends Event
+{
+
+    /**
+     * @var array
+     */
+    private $actions = array();
+
+    /**
+     * @var Translator
+     */
+    private $translator;
+
+    /**
+     * @param Translator $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * Adds an action to the list of available .
+     *
+     * @param string $key    - a unique identifier; it is recommended that it be namespaced i.e. lead.action
+     * @param array  $action - can contain the following keys:
+     *  'label'           => (required) what to display in the list
+     *  'description'     => (optional) short description of event
+     *  'template'        => (optional) template to use for the action's HTML in the company builder
+     *      i.e AcmeMyBundle:CompanyAction:theaction.html.php
+     *  'formType'        => (optional) name of the form type SERVICE for the action; will use a default form with company change only
+     *  'formTypeOptions' => (optional) array of options to pass to formType
+     *  'callback'        => (optional) callback function that will be passed when the action is triggered; return true to
```

Comment:
Let's not support a `callback` here as we are currently refactoring all our other similar uses away from `callback` to `eventName` and dispatch the event instead.